__tests__/fdir.test.ts

```diff
@@ -486,3 +491,27 @@ test(`do not convert \\\\ to \\`, async (t) => {
     "\\\\wsl.localhost\\Ubuntu\\home\\"
   );
 });
+
+test("interrupted iterator should stop yielding results", async (t) => {
+  const api = new fdir().crawl("./src");
+  const iterator = api.withIterator()[Symbol.asyncIterator]();
+  const results: string[] = [];
+  let next = await iterator.next();
+  do {
+    results.push(next.value);
+    iterator.return?.();
+  } while (next.done !== false);
+  t.expect(results.length).toBe(1);
+});
+
+test("aborted iterator should stop yielding results", async (t) => {
+  const aborter = new AbortController();
+  const api = new fdir().withAbortSignal(aborter.signal).crawl("./src");
+  const iterator = api.withIterator();
+  const results: string[] = [];
+  for await (const value of iterator) {
+    results.push(value);
+    aborter.abort();
+  }
+  t.expect(results.length).toBe(1);
+});
```

Comment:
this should detect the presence of `AbortController` just like how it's done in afde92a
```suggestion
if ("AbortController" in globalThis) {
  test("aborted iterator should stop yielding results", async (t) => {
    const aborter = new AbortController();
    const api = new fdir().withAbortSignal(aborter.signal).crawl("./src");
    const iterator = api.withIterator();
    const results: string[] = [];
    for await (const value of iterator) {
      results.push(value);
      aborter.abort();
    }
    t.expect(results.length).toBe(1);
  });
}
```